### PR TITLE
Pair setup change

### PR DESCRIPTION
**Pull request by Apollon77**
This PR addresses #31 by adding support to optionally provide the PairMethod and also ParingTypeFlags available when using PairSetup method.

My Hue is pairing fine.

I hope I did anything correct in Typescript

### CODE REVIEW

File: examples/ip/identify.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  console.log('Found device!');
+  console.log('Found device! ', service);
 
   const client = new HttpClient(service.id, service.address, service.port);
   client
     .identify()
-    .then(() => console.log('Done!'))
-    .catch((e) => console.error(e));
+    .then(() => console.log(service.name, ': Done!'))
+    .catch((e) => console.error(service.name, ': ', e));
```
Comment (mrstegeman) [inline examples/ip/identify.js:12]:
```suggestion
    .then(() => console.log(`${service.name}: Done!`))
    .catch((e) => console.error(`${service.name}:`, e));
```

File: examples/ip/identify.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
-const { HttpClient, IPDiscovery } = require('hap-controller');
-
+//const { HttpClient, IPDiscovery } = require('hap-controller');
+const { HttpClient, IPDiscovery } = require('../../lib/index');
 const discovery = new IPDiscovery();
 
 discovery.on('serviceUp', (service) => {
-  console.log('Found device!');
+  console.log('Found device! ', service);
```
Comment (mrstegeman) [inline examples/ip/identify.js:6]:
```suggestion
  console.log(`Found device: ${service.name}`);
```

File: examples/ip/pair-setup.js [hunk 3]
```diff
@@ -1,19 +1,19 @@
-const { HttpClient, IPDiscovery } = require('hap-controller');
+const { HttpClient, IPDiscovery, PairingMethods } = require('hap-controller');
 
 const discovery = new IPDiscovery();
 
 const pin = 'XXX-YY-ZZZ';
 
 discovery.on('serviceUp', (service) => {
-  console.log('Found device!');
+  console.log('Found device! ', service);
```
Comment (mrstegeman) [inline examples/ip/pair-setup.js:8]:
```suggestion
  console.log(`Found device: ${service.name}`);
```

File: examples/ip/pair-setup.js [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
 
   const client = new HttpClient(service.id, service.address, service.port);
   client
-    .pairSetup(pin)
+    .pairSetup(pin, PairingMethods.PairSetup, 0)
     .then(() => {
-      console.log('Paired! Keep the following pairing data safe:');
+      console.log(service.name, ' Paired! Keep the following pairing data safe:');
```
Comment (mrstegeman) [inline examples/ip/pair-setup.js:14]:
```suggestion
      console.log(`${service.name} paired! Keep the following pairing data safe:`);
```

File: examples/ip/pair-setup.js [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
       console.log(JSON.stringify(client.getLongTermData(), null, 2));
     })
-    .catch((e) => console.error(e));
+    .catch((e) => console.error(service.name, ' ', e));
```
Comment (mrstegeman) [inline examples/ip/pair-setup.js:17]:
```suggestion
    .catch((e) => console.error(`${service.name}:`, e));
```

File: examples/ip/pair-setupauth-prompt.js [hunk 6]
```diff
@@ -8,14 +8,14 @@ const { HttpClient, IPDiscovery } = require('../../lib');
 const discovery = new IPDiscovery();
 
 discovery.on('serviceUp', async (service) => {
-  console.log('Found device!');
+  console.log('Found device! ', service.name);
```
Comment (mrstegeman) [inline examples/ip/pair-setupauth-prompt.js:11]:
```suggestion
  console.log(`Found device: ${service.name}`);
```

File: examples/ip/pair-setupauth-prompt.js [hunk 7]
```diff
@@ -8,14 +8,14 @@ const { HttpClient, IPDiscovery } = require('../../lib');
...
 
   const client = new HttpClient(service.id, service.address, service.port);
 
   const data = await client.startPairing();
   readline.question('Enter PIN: ', async (pin) => {
     await client.finishPairing(data, pin);
-    console.log('Paired! Keep the following pairing data safe:');
+    console.log(service.name, ' Paired! Keep the following pairing data safe:');
```
Comment (mrstegeman) [inline examples/ip/pair-setupauth-prompt.js:18]:
```suggestion
    console.log(`${service.name} paired! Keep the following pairing data safe:`);
```

File: examples/ip/pair-setupauth.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+const { HttpClient, IPDiscovery, PairingMethods } = require('hap-controller');
+
+const discovery = new IPDiscovery();
+
+const pin = 'XXX-YY-ZZZ';
+
+discovery.on('serviceUp', (service) => {
+  console.log('Found device! ', service.name);
```
Comment (mrstegeman) [inline examples/ip/pair-setupauth.js:8]:
```suggestion
  console.log(`Found device: ${service.name}`);
```

File: examples/ip/identify.js [hunk 9]
```diff
@@ -1,14 +1,14 @@
-const { HttpClient, IPDiscovery } = require('hap-controller');
-
+//const { HttpClient, IPDiscovery } = require('hap-controller');
+const { HttpClient, IPDiscovery } = require('../../lib/index');
```
Comment (mrstegeman) [inline examples/ip/identify.js:2]:
```suggestion
const { HttpClient, IPDiscovery } = require('hap-controller');
```

File: examples/ip/pair-setupauth.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+  const client = new HttpClient(service.id, service.address, service.port);
+  client
+    .pairSetup(pin/*, PairingMethods.PairSetupWithAuth*/)
+    .then(() => {
+      console.log(service.name, ' Paired! Keep the following pairing data safe:');
```
Comment (mrstegeman) [inline examples/ip/pair-setupauth.js:14]:
```suggestion
      console.log(`${service.name} paired! Keep the following pairing data safe:`);
```

File: examples/ip/pair-setupauth.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+      console.log(JSON.stringify(client.getLongTermData(), null, 2));
+    })
+    .catch((e) => console.error(service.name, ' ', e));
```
Comment (mrstegeman) [inline examples/ip/pair-setupauth.js:17]:
```suggestion
    .catch((e) => console.error(`${service.name}:`, e));
```

File: examples/ip/remove-pairing.js [hunk 12]
```diff
@@ -11,13 +11,13 @@ const pairingData = {
 };
 
 discovery.on('serviceUp', (service) => {
-  console.log('Found device!');
+  console.log(service.name, ' Found device!');
```
Comment (mrstegeman) [inline examples/ip/remove-pairing.js:14]:
```suggestion
  console.log(`Found device: ${service.name}`);
```

File: examples/ip/remove-pairing.js [hunk 13]
```diff
@@ -11,13 +11,13 @@ const pairingData = {
...
 
   const client = new HttpClient(service.id, service.address, service.port, pairingData);
 
   client
     .removePairing(client.pairingProtocol.iOSDevicePairingID)
-    .then(() => console.log('Done!'))
-    .catch((e) => console.error(e));
+    .then(() => console.log(service.name, ' Done!'))
```
Comment (mrstegeman) [inline examples/ip/remove-pairing.js:20]:
```suggestion
    .then(() => console.log(`${service.name}: done!`))
```

File: examples/ip/remove-pairing.js [hunk 14]
```diff
@@ -11,13 +11,13 @@ const pairingData = {
...
+    .catch((e) => console.error(service.name, ' ', e));
```
Comment (mrstegeman) [inline examples/ip/remove-pairing.js:21]:
```suggestion
    .catch((e) => console.error(`${service.name}:`, e));
```

**Review by mrstegeman — CHANGES_REQUESTED** [review]
Thanks for doing this! I've left some suggestions, but nothing major. There are some style issues as well, which can be fixed with `npm run prettier`

**Comment by Apollon77** [comment]
Sure, template strings are much beter, I just "followed the style" you used in one of the exmaple files :-)

**Comment by Apollon77** [comment]
I also let prettify run and fixed eslint stuuff again :-)

**Review by mrstegeman — APPROVED** [review]

**Pull request merged**